processing/src/main/java/org/apache/druid/segment/virtual/ExpressionSelectors.java

```diff
@@ -518,6 +513,20 @@ private static Object coerceListDimToStringArray(List val)
     return new String[]{null};
   }
 
+  /**
+   * Coerces {@link ExprEval} value back to selector friendly {@link List} if the evaluated expression result is an
+   * array type
+   */
+  public static Object coerceEvalToSelectorObject(ExprEval eval)
+  {
+    if (eval.isArray()) {
```

Comment:
Is it fair to cast all arrays to lists of strings?

Do we still get the behavior we want if we cast them to lists of whatever they really are? (I'm thinking this could make us more future-proof to situations where we support numeric arrays at the storage layer.)

Reply:
i think if we just preserved the base type and covert to a list of that instead of converting it all to a list of strings, that it would still function more or less correctly because the values would be translated to strings further down the chain when reading into the row. Alternatively, an output `ValueType` could be specified to determine how to translate the evaluated expression? But I think just converting to a list as is might be better if it works ok.

Reply:
If converting to a list of the base type works, let's do that, since it's nicer.

My goal here (and w/ the other comment) is to avoid making too many places that encode a lists-should-always-be-strings assumption. We might want to expand that later.